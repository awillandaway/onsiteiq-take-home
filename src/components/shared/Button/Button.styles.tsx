/* eslint-disable import/no-extraneous-dependencies */
import styled, { css } from 'styled-components';
import { lighten } from 'polished';
import { onsiteIqPurple } from 'constants/colors';
import type { ButtonVariant } from './Button';

interface ButtonWrapperProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  variant: ButtonVariant;
}

const primaryColor = '#185818';
const dangerColor = '#851e1e';

export const ButtonWrapper = styled.button<ButtonWrapperProps>`
  min-height: 40px;
  cursor: pointer;
  color: white;
  border: none;
  border-radius: 2px;
  font-size: 16px;
  padding: 4px 12px;

  ${(props) => {
    switch (props.variant) {
      case 'primary':
        return css`
          background-color: ${primaryColor};
          &:hover {
            background-color: ${lighten(0.05, primaryColor)};
          }
        `;
      case 'danger':
        return css`
          background-color: ${dangerColor};
          &:hover {
            background-color: ${lighten(0.05, dangerColor)};
          }
        `;
      case 'secondary':
      default:
        return css`
          background-color: ${onsiteIqPurple};
          &:hover {
            background-color: ${lighten(0.05, onsiteIqPurple)};
          }
        `;
    }
  }};
`;
