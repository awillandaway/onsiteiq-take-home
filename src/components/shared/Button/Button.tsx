import { ButtonWrapper } from './Button.styles';

export type ButtonVariant = 'primary' | 'secondary' | 'danger' | 'default';

interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  text: string;
  variant?: ButtonVariant;
}

export const Button = ({ text, onClick, variant = 'default' }: ButtonProps) => (
  <ButtonWrapper type="button" onClick={onClick} variant={variant}>
    {text}
  </ButtonWrapper>
);
