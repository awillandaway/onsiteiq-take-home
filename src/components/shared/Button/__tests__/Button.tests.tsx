import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';

describe('Button', () => {
  it('Renders correctly', () => {
    const button = renderer.create(<Button text="Test content" />).toJSON();
    expect(button).toMatchSnapshot();
  });

  it('Correctly calls click handler', () => {
    const mockOnClick = jest.fn();
    const { getByRole } = render(<Button text="Test content" onClick={mockOnClick} />);

    fireEvent.click(getByRole('button'));

    expect(mockOnClick).toHaveBeenCalledTimes(1);
  });
});
