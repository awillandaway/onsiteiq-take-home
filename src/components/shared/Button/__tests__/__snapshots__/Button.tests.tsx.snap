// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Renders correctly 1`] = `
<button
  className="sc-bcXHqe jWfLtu"
  type="button"
>
  Test content
</button>
`;
